tensorflow/compiler/tf2xla/frontend_attributes_util.cc

```diff
@@ -0,0 +1,43 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#include "tensorflow/compiler/tf2xla/frontend_attributes_util.h"
+
+#include "tensorflow/core/framework/node_def_util.h"
+#include "tensorflow/core/lib/core/errors.h"
+
+namespace tensorflow {
+
+namespace {
+const char kFrontendAttributesAttribute[] = "_XlaFrontendAttributes";
+}  // namespace
+
+xla::StatusOr<absl::optional<xla::FrontendAttributes>>
+GetFrontendAttributesFromNodeDef(const NodeDef& node_def) {
+  if (!HasNodeAttr(node_def, kFrontendAttributesAttribute)) {
+    return absl::optional<xla::FrontendAttributes>();
```

Comment:
Can you return `absl::nullopt`?

Reply:
I can't just do `return absl::nullopt;` because the `optional` is embedded inside a `StatusOr`:
```
ERROR: tensorflow/compiler/tf2xla/frontend_attributes_util.cc:29:12: error: could not convert 'absl::nullopt' from 'absl::internal::identity_t<absl::nullopt_t> {aka const absl::nullopt_t}' to 'xla::StatusOr<absl::optional<xla::FrontendAttributes> > {aka stream_executor::port::StatusOr<absl::optional<xla::FrontendAttributes> >}'
ERROR:      return absl::nullopt;
```
The alternative is to do: `return xla::StatusOr<absl::optional<xla::FrontendAttributes>>(absl::nullopt);` but I think it ends up being the same as `return absl::optional<xla::FrontendAttributes>();` 

https://github.com/abseil/abseil-cpp/blob/master/absl/types/optional.h#L130

I'll change it anyway if you think it's more obvious with the `nullopt`